src/components/Header/Header.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { memo, useEffect, useRef } from 'react';
+import cn from 'classnames';
+
+type Props = {
+  newTitle: string;
+  setNewTitle: (title: string) => void;
+  onAddNewTodo: (event: React.FormEvent) => void;
+  isAdding: boolean;
+  onChangeAllTodos: () => void;
+  isAllTodosCompleted: boolean;
+};
+
+export const Header: React.FC<Props> = memo((props) => {
+  const {
+    newTitle,
+    setNewTitle,
+    onAddNewTodo,
+    isAdding,
+    onChangeAllTodos,
+    isAllTodosCompleted,
+  } = props;
```

Comment:
```suggestion
export const Header: React.FC<Props> = memo(({ 
  newTitle,
  setNewTitle,
  onAddNewTodo,
  isAdding,
  onChangeAllTodos,
  isAllTodosCompleted,
}) => {
```
You can use destructurization here

Reply:
done